db/schema_tables.cc

```diff
@@ -1868,6 +1902,45 @@ std::vector<mutation> make_drop_function_mutations(shared_ptr<cql3::functions::u
     return make_drop_function_mutations(functions(), *func, timestamp);
 }
 
+/*
+ * UDA metadata serialization/deserialization
+ */
+
+static std::pair<mutation, clustering_key> get_mutation(schema_ptr s, const cql3::functions::user_aggregate& aggregate) {
+    auto name = aggregate.name();
+    auto pkey = partition_key::from_singular(*s, name.keyspace);
+
+    list_type_impl::native_type arg_types;
+    for (const auto& arg_type : aggregate.arg_types()) {
+        arg_types.emplace_back(arg_type->as_cql3_type().to_string());
+    }
+    auto arg_list_type = list_type_impl::get_instance(utf8_type, false);
+    data_value arg_types_val = make_list_value(arg_list_type, std::move(arg_types));
+    auto ckey = clustering_key::from_exploded(
+            *s, {utf8_type->decompose(name.name), arg_list_type->decompose(arg_types_val)});
+    mutation m{s, pkey};
+    return {std::move(m), std::move(ckey)};
+}
+
+std::vector<mutation> make_create_aggregate_mutations(shared_ptr<cql3::functions::user_aggregate> aggregate, api::timestamp_type timestamp) {
+    schema_ptr s = aggregates();
+    auto p = get_mutation(s, *aggregate);
+    mutation& m = p.first;
+    clustering_key& ckey = p.second;
+
+    m.set_clustered_cell(ckey, "final_func", aggregate->finalfunc().name().name, timestamp);
+    data_type state_type = aggregate->sfunc().arg_types()[0];
+    m.set_clustered_cell(ckey, "initcond", state_type->to_string(*aggregate->initcond()), timestamp);
+    m.set_clustered_cell(ckey, "return_type", aggregate->return_type()->as_cql3_type().to_string(), timestamp);
+    m.set_clustered_cell(ckey, "state_func", aggregate->sfunc().name().name, timestamp);
+    m.set_clustered_cell(ckey, "state_type", state_type->as_cql3_type().to_string(), timestamp);
+    return {m};
+}
+
+std::vector<mutation> make_drop_aggregate_mutations(shared_ptr<cql3::functions::user_aggregate> aggregate, api::timestamp_type timestamp) {
+    return make_drop_function_mutations(aggregates(), *aggregate, timestamp);
```

Comment:
I think we need to check that we don't drop a scalar function if it's used in an aggregate.

Reply:
Indeed, I also need to add lots of test cases, including ones that drop and recreate state and final functions.